### ISSUE TITLE
linker command failed on macOS Big Sur

### ISSUE THREAD

**Issue by finddit** [build2/build2#160]
Hi

I am fed up with cmake so decided to try/test build2. 

Followed the steps in the documentation.

Unfortunately the install/build failed with the following error:

```
clang++ -std=c++1y   -o build2/b-boot build2/b-options.b.o build2/b.b.o build2/types-parsers.b.o libbuild2/adhoc-rule-buildscript.b.o libbuild2/adhoc-rule-cxx.b.o libbuild2/algorithm.b.o libbuild2/buildspec.b.o libbuild2/context.b.o libbuild2/depdb.b.o libbuild2/diagnostics.b.o libbuild2/dump.b.o libbuild2/file.b.o libbuild2/filesystem.b.o libbuild2/function.b.o libbuild2/functions-builtin.b.o libbuild2/functions-filesystem.b.o libbuild2/functions-name.b.o libbuild2/functions-path.b.o libbuild2/functions-process-path.b.o libbuild2/functions-process.b.o libbuild2/functions-project-name.b.o libbuild2/functions-regex.b.o libbuild2/functions-string.b.o libbuild2/functions-target-triplet.b.o libbuild2/lexer.b.o libbuild2/module.b.o libbuild2/name.b.o libbuild2/operation.b.o libbuild2/parser.b.o libbuild2/prerequisite.b.o libbuild2/recipe.b.o libbuild2/rule.b.o libbuild2/scheduler.b.o libbuild2/scope.b.o libbuild2/search.b.o libbuild2/target.b.o libbuild2/token.b.o libbuild2/utility-installed.b.o libbuild2/utility-uninstalled.b.o libbuild2/utility.b.o libbuild2/variable.b.o libbuild2/script/builtin-options.b.o libbuild2/script/lexer.b.o libbuild2/script/parser.b.o libbuild2/script/regex.b.o libbuild2/script/run.b.o libbuild2/script/script.b.o libbuild2/script/token.b.o libbuild2/build/script/lexer.b.o libbuild2/build/script/parser.b.o libbuild2/build/script/runner.b.o libbuild2/build/script/script.b.o libbuild2/build/script/token.b.o libbuild2/config/functions.b.o libbuild2/config/init.b.o libbuild2/config/module.b.o libbuild2/config/operation.b.o libbuild2/config/utility.b.o libbuild2/dist/init.b.o libbuild2/dist/module.b.o libbuild2/dist/operation.b.o libbuild2/dist/rule.b.o libbuild2/test/script/lexer.b.o libbuild2/test/script/parser.b.o libbuild2/test/script/runner.b.o libbuild2/test/script/script.b.o libbuild2/test/script/token.b.o libbuild2/test/common.b.o libbuild2/test/init.b.o libbuild2/test/operation.b.o libbuild2/test/rule.b.o libbuild2/test/target.b.o libbuild2/install/functions.b.o libbuild2/install/init.b.o libbuild2/install/operation.b.o libbuild2/install/rule.b.o libbuild2/bin/functions.b.o libbuild2/bin/guess.b.o libbuild2/bin/init.b.o libbuild2/bin/rule.b.o libbuild2/bin/target.b.o libbuild2/bin/utility.b.o libbuild2/c/init.b.o libbuild2/cc/common.b.o libbuild2/cc/compile-rule.b.o libbuild2/cc/gcc.b.o libbuild2/cc/guess.b.o libbuild2/cc/init.b.o libbuild2/cc/install-rule.b.o libbuild2/cc/lexer.b.o libbuild2/cc/link-rule.b.o libbuild2/cc/module.b.o libbuild2/cc/msvc.b.o libbuild2/cc/parser.b.o libbuild2/cc/pkgconfig.b.o libbuild2/cc/target.b.o libbuild2/cc/utility.b.o libbuild2/cc/windows-manifest.b.o libbuild2/cc/windows-rpath.b.o libbuild2/cxx/init.b.o libbuild2/cxx/target.b.o libbuild2/version/init.b.o libbuild2/version/module.b.o libbuild2/version/rule.b.o libbuild2/version/snapshot-git.b.o libbuild2/version/snapshot.b.o libbuild2/version/utility.b.o libbuild2/in/init.b.o libbuild2/in/rule.b.o libbuild2/in/target.b.o ../libbutl/libbutl/b.b.o ../libbutl/libbutl/backtrace.b.o ../libbutl/libbutl/base64.b.o ../libbutl/libbutl/builtin-options.b.o ../libbutl/libbutl/builtin.b.o ../libbutl/libbutl/command.b.o ../libbutl/libbutl/curl.b.o ../libbutl/libbutl/default-options.b.o ../libbutl/libbutl/diagnostics.b.o ../libbutl/libbutl/fdstream.b.o ../libbutl/libbutl/filesystem.b.o ../libbutl/libbutl/git.b.o ../libbutl/libbutl/manifest-parser.b.o ../libbutl/libbutl/manifest-rewriter.b.o ../libbutl/libbutl/manifest-serializer.b.o ../libbutl/libbutl/openssl.b.o ../libbutl/libbutl/pager.b.o ../libbutl/libbutl/path-pattern.b.o ../libbutl/libbutl/path.b.o ../libbutl/libbutl/process-io.b.o ../libbutl/libbutl/process-run.b.o ../libbutl/libbutl/process.b.o ../libbutl/libbutl/project-name.b.o ../libbutl/libbutl/prompt.b.o ../libbutl/libbutl/regex.b.o ../libbutl/libbutl/semantic-version.b.o ../libbutl/libbutl/sendmail.b.o ../libbutl/libbutl/sha1.b.o ../libbutl/libbutl/sha256.b.o ../libbutl/libbutl/standard-version.b.o ../libbutl/libbutl/string-parser.b.o ../libbutl/libbutl/tab-parser.b.o ../libbutl/libbutl/target-triplet.b.o ../libbutl/libbutl/timestamp.b.o ../libbutl/libbutl/unicode.b.o ../libbutl/libbutl/utility.b.o ../libbutl/libbutl/uuid-freebsd.b.o ../libbutl/libbutl/uuid-io.b.o ../libbutl/libbutl/uuid-linux.b.o ../libbutl/libbutl/uuid-macos.b.o ../libbutl/libbutl/uuid-windows.b.o ../libbutl/libbutl/uuid.b.o ../libbutl/libbutl/win32-utility.b.o -lpthread
Undefined symbols for architecture x86_64:
  "std::__1::__match_any_but_newline<build2::script::regex::line_char>::__exec(std::__1::__state<build2::script::regex::line_char>&) const", referenced from:
      vtable for std::__1::__match_any_but_newline<build2::script::regex::line_char> in run.b.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [build2/b-boot] Error 1
```

**Comment by finddit** [comment]
Same results by bootstrapping using the following documentation:
https://build2.org/build2-toolchain/doc/build2-toolchain-install.xhtml#bootstrap-macosx

clang version:

```
 clang++ --version         
Apple clang version 12.0.5 (clang-1205.0.22.11)
Target: x86_64-apple-darwin20.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

All help is much appreciated.

**Comment by boris-kolpackov** [comment]
Can you try the latest staged version of the toolchain: https://build2.org/community.xhtml#stage

For background, there is a bug in `libc++` regex that we provide a work around for. Hoping that the Clang folks will fix it in the next release, we only enable it for the current version. But it seems there is no urgency in fixing it so now we've enabled this workaround for all future versions.

**Comment by jwheeler31** [comment]
TL;DR:
```
$ https://stage.build2.org/0/0.14.0-a.0/build2-install-0.14.0-a.0-stage.sh
$ chmod +x build2-install-0.14.0-a.0-stage.sh
$ ./build2-install-0.14.0-a.0-stage.sh
```
I experienced similar behavior on: `macOS Big Sur, Version 11.5.1`.

@boris-kolpackov, just successfully finished the install - will update with any further issues.

**Comment by springmeyer** [comment]
Thank you for the TLDR @jwheeler31, that got things working for me with `macOS Big Sur, Version 11.6 (20G165)`
